nmtwizard/preprocess/operators/noise.py

```diff
@@ -100,6 +132,40 @@ def _apply_word_noise(self, tokens):
             if not token.is_placeholder():
                 if self._drop_word_prob > 0 and random.random() <= self._drop_word_prob:
                     continue
+                # TODO : joiners
```

Comment:
Joiners are not an issue here since the code is manipulating token objects. So you can remove this TODO.

Reply:
I was thinking about the case where we swap tokens for example. But since we are inserting noise anyway, maybe it's not a big deal.  I removed the comment.

Reply:
Ah, I see. But yes, it's probably not worth it to check if the tokens are joined or not. It will just make the logic more complex.